src/System.Private.CoreLib/shared/System/Diagnostics/Contracts/Contracts.cs

```diff
@@ -466,7 +467,7 @@ public static void Ensures(bool condition, string userMessage)
         /// This method can only be used within the argument to the <seealso cref="Ensures(bool)"/> contract.
         /// </remarks>
         [Pure]
-        public static T ValueAtReturn<T>(out T value) { value = default; return value; }
+        public static T ValueAtReturn<T>(out T value) { value = default!; return value; }
```

Comment:
won't this always be null for reference types?

Reply:
ah forgot - generics..

Reply:
It's also not the actual implementation that ends up being used.  This is only used as part of a Contract.Ensures, and the contract rewriter replaces this call with the actual result of the method.  This body is just a dummy stand-in.